src/ripple/consensus/Consensus.h

```diff
@@ -653,11 +653,31 @@ Consensus<Derived, Traits>::startRound(
     {
         prevCloseTime_ = rawCloseTimes_.self;
     }
+
+    Mode startMode = proposing ? Mode::proposing : Mode::observing;
+
+    // We were handed the wrong ledger
+    if (prevLedger.id() != prevLedgerID)
+    {
+        // try to acquire the correct one
+        if(auto newLedger = impl().acquireLedger(prevLedgerID))
+        {
+            prevLedger = *newLedger;
```

Comment:
Ledger_t looks like it's usually a thin wrapper around a shared_ptr (i.e. RCLCxLedger). There's a minor benefit if we move here instead of copy.